### PR TITLE
Open port 80 for Let's encrypt validation to pass. Move away from the…

### PR DESCRIPTION
**Pull request by mclueppers**
… hardcoded 10.0.0.0/8 network range to proper list

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -162,15 +162,23 @@ resource "aws_security_group" "pritunl" {
     from_port   = 22
     to_port     = 22
     protocol    = "tcp"
-    cidr_blocks = ["10.0.0.0/8"]
+    cidr_blocks = ["${split(",", join(",", var.internal_cidrs))}"]
```
Comment (leventyalcin) [inline main.tf]: The variable `internal_cidrs`is a list already. I think `join`-ing the list and `split` it back is overhead at this point. I think `var.internal_cidrs` is enough in here.

File: main.tf [hunk 2]
```diff
@@ -162,15 +162,23 @@ resource "aws_security_group" "pritunl" {
...
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  # HTTPS access
   ingress {
     from_port   = 443
     to_port     = 443
     protocol    = "tcp"
-    cidr_blocks = ["10.0.0.0/8"]
+    cidr_blocks = ["${split(",", join(",", var.internal_cidrs))}"]
```
Comment (leventyalcin) [inline main.tf]: The same with above.

File: main.tf [hunk 3]
```diff
@@ -186,7 +194,7 @@ resource "aws_security_group" "pritunl" {
     from_port   = -1
     to_port     = -1
     protocol    = "icmp"
-    cidr_blocks = ["10.0.0.0/8"]
+    cidr_blocks = ["${split(",", join(",", var.internal_cidrs))}"]
```
Comment (leventyalcin) [inline main.tf]: The same with above.

File: main.tf [hunk 4]
```diff
@@ -212,26 +220,29 @@ resource "aws_security_group" "allow_from_office" {
...
 
-  # HTTP access
+  # HTTPS access
   ingress {
+    description = "Allow HTTPS access from select CIDRs"
     from_port   = 443
     to_port     = 443
     protocol    = "tcp"
-    cidr_blocks = ["${var.whitelist}"]
+    cidr_blocks = ["${split(",", join(",", var.whitelist))}"]
```
Comment (leventyalcin) [inline main.tf]: `whitelist` is a list too. So, I think does not need to `join` and `split`.

File: main.tf [hunk 5]
```diff
@@ -212,26 +220,29 @@ resource "aws_security_group" "allow_from_office" {
...
   }
 
   # ICMP
   ingress {
+    description = "Allow ICMPv4 from select CIDRs"
     from_port   = -1
     to_port     = -1
     protocol    = "icmp"
-    cidr_blocks = ["${var.whitelist}"]
+    cidr_blocks = ["${split(",", join(",", var.whitelist))}"]
```
Comment (leventyalcin) [inline main.tf]: `whitelist` is a list too. So, I think does not need to `join` and `split`.

**Comment by leventyalcin** [comment]
Hey @mclueppers, 

I noticed that there is no change for opening port 80. 

Also, I was checking if there are fixed IP addresses for checks, but, unfortunately, there is none. 

```
Let’s Encrypt IPs
Let’s Encrypt will validate from a number of different IP addresses in the future, and will not announce which ones in advance. You should make sure your validation server is available to all IPs.

Some people who are issuing for non-HTTP services want to avoid exposing port 80 to anyone except Let’s Encrypt’s validation server. If you’re in that category you may want to use the DNS challenge instead.
```

Maybe it's better to leave that responsibility to the users. They can add `0.0.0.0/0` to the whitelist until they run checks and they can remove afterward if they want. 

As you said, it's better if we add that as well. 

Cheers,
L.

File: main.tf (hunk 1 above)
Comment (mclueppers) [reply main.tf]: I guess it already has been mitigated in later version but I've got this workaround from the following issue. I can always revert it back if you deem it appropriate

https://github.com/hashicorp/terraform/issues/13103

**Comment by mclueppers** [comment]
In regard to Let's encrypt I don't see an issue with having port 80 wide open. pritunl is forwarding HTTP -> HTTPS once you have your DNS name properly set in the configuration of the server. There are some suggestions that one can easily utilise the Nginx/Apache IP filters and allow HTTP access to /.well-known/challenge. It's up for a discussion. 
As for the DNS challenge - where do we get it from so we can update our DNS records? There is an admin effort required to achieve that whilst HTTP just work out of the box :)

**Comment by mclueppers** [comment]
Any chance to get this merged? :)

**Pull request merged**